### PR TITLE
Image grid: softer subset verification

### PR DESCRIPTION
**Pull request by markotoplak**
##### Issue
Conditions for valid subsets are needlessly strong. I noticed this when fixing something else.

##### Description of changes
Subset computation could be based on ids alone (as elsewhere), or, optionally here, on image names. The active subset is a union of ids and matching image names.

Still warns the user if an element from subset can not be found in the data.

##### Includes
- [X] Code changes
- [X] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by markotoplak** [comment]
We discussed this with @janezd in the morning. Could either you, @janezd, or perhaps @PrimozGodec look at this?

**Review by janezd — COMMENTED** [review]
I started reviewing and commented that I don't understand if there is a reason to iterating over `self.data`, which constructs `RowInstance`'s, instead of iterating over `self.data.ids`. I wanted to refactor the code in this way, but then discovered that the code collects id's and then matches them back, which seems redundant.

I wrote an alternative implementation. Is it OK or have I overlooked something?

File: orangecontrib/imageanalytics/widgets/owimagegrid.py [hunk 1]
```diff
@@ -328,27 +331,35 @@ def set_pixmap(future, thumb=thumbnail):
...
+        self.Warning.extras_in_subset.clear()
+        if not (self.data and self.data_subset):
+            return None
 
-        if self.data and self.data_subset and len(self.stringAttrs) > 0:
-            transformed = self.data_subset.transform(self.data.domain)
-            attr = self.stringAttrs[self.imageAttr]
+        subset_ids = set(self.data_subset.ids)
+        data_is_subset = np.array(
+            [id_ in subset_ids for id_ in self.data.ids if id_ in self.data.ids])
```
Comment (markotoplak) [inline orangecontrib/imageanalytics/widgets/owimagegrid.py:345]: The second part `if id_ in self.data.ids` does not seem like needed. :)

**Comment by markotoplak** [comment]
@janezd, thanks, looks good except for that small inefficiency. :)

File: orangecontrib/imageanalytics/widgets/owimagegrid.py (hunk 1 above)
Comment (janezd) [reply orangecontrib/imageanalytics/widgets/owimagegrid.py]: Better safe than sorry.

**Pull request merged**